### ISSUE TITLE
Enhancement: Set timestamp in log

### ISSUE THREAD

**Issue by goldeelox** [anshooarora/extentreports#301]
I've been using a modified version of ExtentReporterNG to generate reports. It works great, but as mentioned in #161, the log timestamps are all recorded as the time at which the report is generated.

It would be great to have a Date parameter in the log method to override the timestamp. For now, I'll just hide the column.

Thanks for all the hard work on the project so far!!!

**Comment by anshooarora** [comment]
Its possible using `ITest getTest()`:

```
for (Log log : test.getTest().getLogList()) {
    log.setTimestamp(timestamp);
}
```

But, I can default the log timestamps to the test-start time when used via listener.

**Comment by anshooarora** [comment]
Available by default in 2.40.2-beta1.

**Comment by vinuthareddy** [comment]
Hi,
I am using 2.40.2 version of the extent reports.
But i am still seeing the above mentioned issue.
Timestamp field in the report is getting updated with test end time for all the logsteps.
Please help.
